apiserver/facades/client/spaces/shims.go

```diff
@@ -99,10 +119,34 @@ func (s *stateShim) SubnetByCIDR(cidr string) (networkingcommon.BackingSubnet, e
 	return networkingcommon.NewSubnetShim(result), nil
 }
 
+// MovingSubnet wraps state.Subnet so that it
+// returns the MovingSubnet indirection.
+func (s *stateShim) MovingSubnet(id string) (MovingSubnet, error) {
+	result, err := s.State.Subnet(id)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return &movingSubnet{result}, nil
+}
+
+// AllConstraints returns all constraints in the model,
+// wrapped in the Constraints indirection.
+func (s *stateShim) AllConstraints() ([]Constraints, error) {
```

Comment:
I wonder if this is actually a shim, but rather a transformational layer between boundary layers. I think this does a dis-service of calling it a shim.